C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operators-using-member-functions.md

```diff
@@ -0,0 +1,106 @@
+# Overloading Unary Operators Using Member Functions  
+
+Unary Operators are operators which operate on one operand. For example: +10, -10, ++i, i++ etc. In this file, we are going to overload minus '-' operator. For every operator to be overloaded, a separate function called **"operator function"** has to be defined. Operator function is a function defined when an operator is to be overloaded. The keyword **'operator'** is used to create operator functions.  
+
+If the operator function is a member function of a class, then the general syntax for declaring a operator function to overload unary operator is as follows:  
+```C++
+class <class_name>
+{
+   public:
+         <return_type> operator opr ();   //prototype
+};
+<return_type> <class_name> :: operator opr ()
+{
+  //operator function definition
+}
+```  
+
+Where,  
+
+***return_type*** -> type of the value returned by the operator function  
+***class_name*** -> name of the class to which the operator function is the member function  
+***operator*** -> keyword used to create an operator function, operator function is used only to overload operators  
+***opr*** -> opr is the unary operator that is being overloaded.  
+***()*** -> empty parentheses list indicates the operator function is the member function of a class used to overload unary operators  
+
+This syntax for overloading the unary operators changes slightly if operator function is a friend to a class.  
+
+Here's a program below explaining the same:  
+```C++
+#include <iostream>
+using namespace std;
+class girlscript
+{
+  private:
+         int a;
+         int b;
+  public:
+  void operator - ();  //prototype of operator function  
+  void display();
+  girlscript()    //constructor
+  {
+    a = 1;
+    b = 2;
+  }
+};
+void girlscript :: operator - () //member operator function definition
+{
+  a = -a;
+  b = -b;
+}
+void girlscript :: display()
+{
+  cout << "a = " << a << endl;
+  cout << "b = " << b << endl;
+}
+int main()
+{
+  girlscript g;
+  g.display();
+  -g;  // trying to use '-' on object
+  g.display();
+}
+```  
+
+### *Output*  
```

Comment:
@124015001 please fix the alignment

![image](https://user-images.githubusercontent.com/84583787/141330712-a0f33dfe-0749-4ec7-8232-c9a05f47acbb.png)

also please add all the output inside ``` highlighter